### PR TITLE
Add save state to stating of the application

### PR DESCRIPTION
**Pull request by MostafaBalata**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -60,19 +60,20 @@ const transactionHandler = async (transaction) => {
...
+    lastBlockNumberFilePath } = command(watchingConfigPath, lastBlockNumber);
   logger.debug('Start process');
 
   const PromisifiedAbiObjects = addresses.map(async address => (
     { address, abi: await getABI(address) }
   ));
 
   (await Promise.all(PromisifiedAbiObjects)).forEach((object) => {
-    addressAbiMap[object.address] = new Decoder(object.abi);
+    addressAbiMap[object.address.toLowerCase()] = new Decoder(object.abi);
```
Comment (Moejoe90) [inline src/index.js]: good catch

File: src/jsonrpc.js [hunk 2]
```diff
@@ -192,6 +194,12 @@ export default class JsonRpc {
           }
           this.currentBlock = parseInt(this.currentBlock, 10) + 1;
           logger.debug(`Current block number is ${this.currentBlock}`);
+
+          // @TODO: Move this writing outside this module
```
Comment (Moejoe90) [inline src/jsonrpc.js]: Move file writing outside of this module

File: src/logger.js [hunk 3]
```diff
@@ -19,10 +19,15 @@ const logger = new (winston.Logger)({
  * @param {*} customMessage 
  */
 export const logError = (error, customMessage = null, isStack = true) => {
-  logger.error(JSON.stringify({ type: 'Error',
-    message: error.message,
-    stack: isStack ? error.stack : null,
-    customMessage }));
+  if (getEnv('OUTPUT_TYPE') === 'terminal') {
```
Comment (Moejoe90) [inline src/logger.js]: Lets make it a switch case to be able to easier add more output modes :)

File: src/utils.js [hunk 4]
```diff
@@ -44,3 +46,9 @@ export const isQueriedTransaction = ({ txn, txnReceipts, logs, addresses }) =>
   !!((isRegularQueriedTransaction({ QueriedAddress: txn.to, addresses })
     || isContractCreationQueriedTransaction(txnReceipts, addresses))
       || isLogsQueriedTransaction({ logCount: logs.length }));
+
+/**
+ * check if path is exists or not
+ * @param string path 
+ */
+export const isPathExist = (path) => fs.existsSync(path)
```
Comment (Moejoe90) [inline src/utils.js]: `doesPathExist` is a better name

File: src/logger.js (hunk 3 above)
Comment (MostafaBalata) [reply src/logger.js]: This is better for this use case and we don't have plan to other format, are we?

**Pull request merged**